docs/metrics.md

```diff
@@ -4,30 +4,173 @@ Let's look at the various metrics you can calculate using Valor.
+&17 \hspace{1em} | \quad | \quad \textbf{if }j\text{ not in }visited\_groundtruths \textbf{ then} \\
+&18 \hspace{1em} | \quad | \quad | \quad visited.add(j) \\
+&19 \hspace{1em} | \quad | \quad | \quad ranked\_ious.append(ious[j]) \\
+&20 \hspace{1em} | \quad | \quad | \quad \textbf{break} \\
+&21 \hspace{1em} | \quad | \quad \textbf{end} \\
+&22 \hspace{1em} | \quad \textbf{end} \\
+&23 \hspace{1em} \textbf{end} \\
+\end{aligned}
+$$
+
+### Precision-Recall Curve
+
+We can now compute the precision-recall curve using our previously ranked IoU's. We do this by iterating through the ranked IoU's and creating points cumulatively using recall and precision.
+
+$$
+\begin{aligned}
+&\underline{\large\textbf{Algorithm 2} \hspace{0.5em} \text{Precision-Recall Curve}} \\
+&\textbf{Data: }\text{Ranked list of IoU's for a label and a IoU threshold between 0 and 1.} \\
+&\textbf{Results: }\text{List of points on a curve.} \\
+\\
+&1 \hspace{1.5em} ranked\_ious \gets \text{list of IoU's} \\
+&2 \hspace{1.5em} threshold \gets \text{IoU Threshold} \\
+\\
+&3 \hspace{1.5em} n \gets \text{total number of groundtruths.} \\
+&4 \hspace{1.5em} count\_tp \gets 0 \\
+\\
+&5 \hspace{1.5em} curve \gets \text{new empty list} \\
+\\
+&6 \hspace{1.5em} \textbf{for }i=0\text{ to }n-1\textbf{ do} \\
+&7 \hspace{1.5em} | \quad \textbf{if } ranked\_ious[i] \ge threshold \textbf{ then} \\
+&8 \hspace{1.5em} | \quad | \quad count\_tp \gets count\_tp + 1 \\
+&9 \hspace{1.5em} | \quad \textbf{end} \\
+&10 \hspace{1em} | \quad precision \gets count\_tp \mathrel{{/}} (i+1) \\
+&11 \hspace{1em} | \quad recall \gets count\_tp \mathrel{{/}} n \\
+&12 \hspace{1em} | \quad curve.append((recall, precision)) \\
+&13 \hspace{1em} \textbf{end} \\
+\end{aligned}
+$$
+
+### Calculating Average Precision
+
+Average precision is defined as the integration of the precision-recall curve. However, due to the varying nature of datasets it has been shown that interpolating this curve with a fixed set of points helps to reduce inconsistencies between dataset splits. The defacto standard has been to use a 101-point interpolation of the precision-recall curve to compute this integral.
+
+$$
+AP = \frac{1}{101} \sum\limits_{r\in\{ 0, 0.01, \ldots , 1 \}}\rho_{interp}(r)
+$$
+
+$$
+\rho_{interp} = \underset{\tilde{r}:\tilde{r} \ge r}{max \ \rho (\tilde{r})}
+$$
+
+### References
+- [MS COCO Detection Evaluation](https://cocodataset.org/#detection-eval)
+- [Mean Average Precision (mAP) Using the COCO Evaluator](https://pyimagesearch.com/2022/05/02/mean-average-precision-map-using-the-coco-evaluator/)
 
-\*When calculating IOUs for object detection metrics, Valor handles the necessary conversion between different types of image annotations. For example, if your model prediction is a polygon and your ground truth is a raster, then the raster will be converted to a polygon prior to calculating the IOU.
\ No newline at end of file
+### Notes
```

Comment:
can you move this up to below the original "Object Detection" section now that you've added more content between the two?